alfresco-search/src/main/resources/solr/instance/templates/noRerank/conf/schema.xml

```diff
@@ -1,766 +1,784 @@
+                   spatialContextFactory="org.locationtech.spatial4j.context.SpatialContextFactory"
+                   distErrPct="0.025"
+                   maxDistErr="0.000009"
+                   distanceUnits="degrees" />
+
+
+      <!-- Suggestion -->
+
+      <fieldType name="text_shingle" class="solr.TextField" positionIncrementGap="100">
+         <analyzer type="index">
+            <tokenizer class="solr.ICUTokenizerFactory"/>
+            <filter class="solr.ICUNormalizer2FilterFactory" name="nfkc_cf" mode="compose" />
+	    <filter class="solr.ShingleFilterFactory" minShingleSize="2" maxShingleSize="3" outputUnigrams="true" outputUnigramsIfNoShingles="true" tokenSeparator=" " />
+         </analyzer>
+         <analyzer type="query">
+            <tokenizer class="solr.ICUTokenizerFactory"/>
+            <filter class="solr.ICUNormalizer2FilterFactory" name="nfkc_cf" mode="compose" />
+	    <filter class="solr.ShingleFilterFactory" minShingleSize="2" maxShingleSize="3" outputUnigrams="true" outputUnigramsIfNoShingles="true" tokenSeparator=" " />
+         </analyzer>
+      </fieldType>
+
+      <fieldType name="text_min_hash" class="solr.TextField" positionIncrementGap="100">
+         <analyzer type="index">
+            <tokenizer class="solr.ICUTokenizerFactory"/>
+            <filter class="solr.ICUNormalizer2FilterFactory" name="nfkc_cf" mode="compose" />
+	    <filter class="solr.ShingleFilterFactory" minShingleSize="5" maxShingleSize="5" outputUnigrams="false" outputUnigramsIfNoShingles="false" tokenSeparator=" " />
+<!--
+	    <filter class="org.apache.lucene.analysis.minhash.ContextAccumulatingFilterFactory" />
+-->
+	    <filter class="org.apache.lucene.analysis.minhash.MinHashFilterFactory" hashCount="1" hashSetSize="1" bucketCount="512" />
+         </analyzer>
+         <analyzer type="query">
+            <tokenizer class="solr.KeywordTokenizerFactory" />
+         </analyzer>
+      </fieldType>
+
+
+       <fieldType name="text_plain" class="solr.TextField" positionIncrementGap="100">
+           <analyzer>
+               <tokenizer class="solr.ICUTokenizerFactory"/>
+           </analyzer>
+       </fieldType>
+
+       <!-- Cross-locale -->
+       <!-- Include word splitting -->
+       <fieldType name="text___" class="solr.TextField" positionIncrementGap="100" autoGeneratePhraseQueries="true">
+           <analyzer type="index">
+               <tokenizer class="solr.ICUTokenizerFactory"/>
+               <filter class="org.apache.solr.analysis.WordDelimiterGraphFilterFactory"
+                       generateWordParts="1"
+                       generateNumberParts="1"
+                       catenateWords="1"
+                       catenateNumbers="1"
+                       catenateAll="1"
+                       splitOnCaseChange="1"
+                       splitOnNumerics="1"
+                       preserveOriginal="1"
+                       stemEnglishPossessive="1"/>
```

Comment:
Do we definitely want to stem English possessives for the cross-language index field? (Were we doing this before?)

Reply:
We didn't. The previous configuration used a symmetric chain (i.e. index analyzer = query analyzer) without any stemming filter.

The stemming filter is strictly tied to a given language, so if this field needs to be cross-language I see some difficulties in applying them here. 

I know, the current synonym filter is using the english synonyms file so even if it is supposed to be part of a cross-language analyzer it has something to do with the english language. My idea is to add several synonyms file to the filter configuration (the synonyms parameter accepts a list of files) once we setup proper synonyms also for other languages. 

If that could work for synonyms, I think it wouldn't for stemming: chaining several stemming filters would produce a mess in the term dictionary.